### ISSUE TITLE
Filter logical operator "OR" not working

### ISSUE THREAD

**Issue by lucianospi** [directus/api#1402]
# Bug Report
This happens with APP 7.10.0 and API 2.6.0.

When using logical operator "OR" with `filter` query parameters in a API request, the response include results satisfying only the first predicate thus ignoring the `OR` and the second part.

# Steps to Reproduce
1. Add a custom collection
2. Create a collection element with property `content_max_age` set to *2020-10-15*
2. Create a collection element with property `content_max_age` left blank (thus saved in DB as *null*)
3. Make an API request as following:
`{{ENDPOINT}}/_/items/banner_item?filter[content_max_age][eq]=2020-10-15&filter[content_max_age][logical]=or&filter[content_max_age][null]`
or 
`{{ENDPOINT}}/_/items/banner_item?filter[content_max_age][eq]=2020-10-15&filter[content_max_age][logical]=or&filter[content_max_age][empty]`
4. Check API response body

## Expected Behaviour
The API response body should include:
1. element created at step 1 (`content_max_age` set to *2020-10-15*)
2. element created at step 2 (`content_max_age` left blank - *null* in database)

## Actual Behaviour
The API response body includes only:
1. element created at step 1 (`content_max_age` set to *2020-10-15*)

## Other Context & Screenshots
![image](https://user-images.githubusercontent.com/56389118/68465308-83ac5e80-0212-11ea-8f13-2723e30c72de.png)

![image](https://user-images.githubusercontent.com/56389118/68464899-bace4000-0211-11ea-9f98-a4e9ad2f3a01.png)

**Comment by itsmerhp** [comment]
Hello @lucianospi 
Please try with below  filters :

**Logical operator for same field :** {{ENDPOINT}}/_/items/banner_item?filter[content_max_age][0][eq]=2020-10-15&filter[content_max_age][1][logical]=or&filter[content_max_age][1][null]

**Logical operator for different fields :** {{ENDPOINT}}/_/items/banner_item?filter[content_max_age][eq]=2020-10-15&filter[id][logical]=OR&filter[id][in]=1,2

**Comment by bjgajjar** [comment]
@lucianospi - May I please have an update on the solution which is provided by @itsmerhp ?

**Comment by bjgajjar** [comment]
Hey, @lucianospi - Can you please provide your inputs on this issue?

**Comment by bjgajjar** [comment]
@lucianospi - 🔔

**Comment by lucianospi** [comment]
![Screenshot 2020-01-08 at 12 46 46](https://user-images.githubusercontent.com/56389118/71976065-b7f1ca80-3215-11ea-892d-48a5bad5fb38.png)
![Screenshot 2020-01-08 at 12 48 49](https://user-images.githubusercontent.com/56389118/71976066-b7f1ca80-3215-11ea-887b-6c54757fe9fd.png)
Hello guys, apologises for the late feedback. I had a go with the suggestions provided by @itsmerhp . Unfortunately none worked.

As you can see in the screenshots, there are existing elements with field "content_max_age" set as null. However that is not returned when using the logical operator OR in the filters.

**Comment by lucianospi** [comment]
Hei guys, got an update on this.

I realises inverting the order of the filter parameters it works! Just as mentioned by @itsmerhp but with a little change:

This call will return valid results:
{{ENDPOINT}}/_/items/banner_item?filter[content_max_age][0][null]&filter[content_max_age][1][logical]=or&filter[content_max_age][1][gte]=2020-02-19

![image](https://user-images.githubusercontent.com/56389118/75046521-84b58300-54c5-11ea-9701-fa346f983a91.png)

while this one will not return results
{{ENDPOINT}}/_/items/banner_item?fields=content_max_age,id&filter[content_max_age][0][eq]=2020-10-15&filter[content_max_age][1][logical]=or&filter[content_max_age][1][null]
![image](https://user-images.githubusercontent.com/56389118/75046422-533cb780-54c5-11ea-93a2-fe150cb6e047.png)

Basically the query param with the NULL value must be the first one of the phrase!

Thank you for all your support!

**Comment by benhaynes** [comment]
Awesome! Can we get this added to the docs?

@directus/api ?
